okhttp/src/main/java/okhttp3/internal/http2/Hpack.java

```diff
@@ -510,8 +512,16 @@ void writeInt(int value, int prefixMask, int bits) {
     }
 
     void writeByteString(ByteString data) throws IOException {
-      writeInt(data.size(), PREFIX_7_BITS, 0);
-      out.write(data);
+      if (useCompression && Huffman.get().encodedLength(data) < data.size()) {
+        Buffer huffmanBuffer = new Buffer();
```

Comment:
If ever we want to really micro-optimize this, we can probably prevent this allocation by reusing the buffer. Not yet!